config/certmanager/certificate.yaml

```diff
@@ -21,4 +21,4 @@ spec:
   issuerRef:
     kind: Issuer
     name: selfsigned-issuer
-  secretName: $(SERVICE_NAME)-cert # this secret will not be prefixed, since it's not managed by kustomize
+  secretName: capm3-webhook-service-cert # this secret will not be prefixed, since it's not managed by kustomize
```

Comment:
is there any reason to hard code it?

Reply:
@lentzi90 had a comment on this other PR about this: https://github.com/kubernetes-sigs/cluster-api/pull/9223#discussion_r1297093508

Basically replacements doesn't work exactly the same way as vars and dynamic values don't work out of the box.